src/main/scala/edu/ie3/powerFactory2psdm/config/ConfigValidator.scala

```diff
@@ -0,0 +1,149 @@
+      .map(conf => conf.conversionMode)
+      .collect {
+        case pvModelGeneration: PvModelGeneration => pvModelGeneration
+      }
+      .map(validatePvModelGenerationParams)
+  }
+
+  def validatePvModelGenerationParams(params: PvModelGeneration): Unit = {
+    validateGenerationMethod(params.albedo, 0, 1) match {
+      case Success(_) =>
+      case Failure(exc: Exception) =>
+        throw ConversionConfigException(
+          s"The albedo of the plants surrounding: ${params.albedo} isn't valid. Exception: ${exc.getMessage}"
+        )
+    }
+    validateGenerationMethod(params.azimuth, -90, 90) match {
+      case Success(_) =>
+      case Failure(exc: Exception) =>
+        throw ConversionConfigException(
+          s"The azimuth of the plant: ${params.azimuth} isn't valid. Exception: ${exc.getMessage}"
+        )
+    }
+    validateGenerationMethod(params.etaConv, 0, 100) match {
+      case Success(_) =>
+      case Failure(exc: Exception) =>
+        throw ConversionConfigException(
+          s"The efficiency of the plants inverter: ${params.azimuth} isn't valid. Exception: ${exc.getMessage}"
+        )
+    }
+    validateGenerationMethod(params.kG, 0, 1) match {
+      case Success(_) =>
+      case Failure(exc: Exception) =>
+        throw ConversionConfigException(
+          s"The PV generator correction factor (kG): ${params.kG} isn't valid. Exception: ${exc.getMessage}"
+        )
+    }
+    validateGenerationMethod(params.kT, 0, 1) match {
+      case Success(_) =>
+      case Failure(exc: Exception) =>
+        throw ConversionConfigException(
+          s"The PV temperature correction factor (kT): ${params.kT} isn't valid. Exception: ${exc.getMessage}"
+        )
+    }
+  }
+
+  def validateGenerationMethod(
+      genMethod: GenerationMethod,
+      lowerBound: Double,
+      upperBound: Double
+  ): Try[Unit] =
+    genMethod match {
+      case Fixed(value) =>
+        checkForBoundViolation(value, lowerBound, upperBound)
+      case UniformDistribution(min, max) =>
+        if (min > max)
+          return Failure(
+            ConversionConfigException(
+              s"The minimum value: $min exceeds the maximum value: $max"
+            )
+          )
+        if (min < lowerBound && max > upperBound)
+          return lowerUpperBoundViolation(min, max, lowerBound, upperBound)
+        else if (min < lowerBound) return lowerBoundViolation(min, lowerBound)
+        else if (max > upperBound) return upperBoundViolation(max, upperBound)
+        Success()
+      case NormalDistribution(mean, _) =>
+        checkForBoundViolation(mean, lowerBound, upperBound)
+    }
+
+  def checkForBoundViolation(
+      value: Double,
+      lowerBound: Double,
+      upperBound: Double
+  ): Try[Unit] = {
+    if (value < lowerBound) return lowerBoundViolation(value, lowerBound)
+    if (value > upperBound) return upperBoundViolation(value, upperBound)
+    Success()
+  }
+
+  def lowerBoundViolation(
```

Comment:
Very nice extraction of methods to avoid code duplicates! 👍🏼 
However, you may consider making these ones `private` as they are not meaningful to be called from the outside, aren't they? 🤔 
Applies for all methods - pls double check which ones really should be `public` and make the others `private` pls